hoomd/Trigger.h

```diff
@@ -87,5 +108,185 @@ class PYBIND11_EXPORT PeriodicTrigger : public Trigger
         uint64_t m_phase;
     };
 
+/** Before trigger
+ *
+ *  Trigger every time step before `m_timestep`.
+*/
+class PYBIND11_EXPORT BeforeTrigger : public Trigger
+    {
+    public:
+
+    BeforeTrigger(uint64_t timestep) : Trigger(), m_timestep(timestep) {}
+
+    bool compute(uint64_t timestep)
+        {
+        return timestep < m_timestep;
+        }
+
+    /// Get the point where triggering will stop
+    uint64_t getTimestep() {return m_timestep;}
+
+    /// Set the point where triggering will stop
+    void setTimestep(uint64_t timestep) {m_timestep = timestep;}
+
+    protected:
+        /// Always trigger timestep this timestep
+        uint64_t m_timestep;
+    };
+
+/** On trigger
+ *
+ *  Trigger on timestep `m_timestep`.
+*/
+class PYBIND11_EXPORT OnTrigger : public Trigger
+    {
+    public:
+
+    OnTrigger(uint64_t timestep) : Trigger(), m_timestep(timestep) {}
+
+    bool compute(uint64_t timestep)
+        {
+        return timestep == m_timestep;
+        }
+
+    /// Get the point where triggering will stop
+    uint64_t getTimestep() {return m_timestep;}
+
+    /// Set the point where triggering will stop
+    void setTimestep(uint64_t timestep) {m_timestep = timestep;}
+
+    protected:
+        /// Always trigger timestep this timestep
+        uint64_t m_timestep;
+    };
+
+/** After trigger
+ *
+ *  Trigger every time step after `m_timestep`.
+*/
+class PYBIND11_EXPORT AfterTrigger : public Trigger
+    {
+    public:
+
+    AfterTrigger(uint64_t timestep) : Trigger(), m_timestep(timestep) {}
+
+    bool compute(uint64_t timestep)
+        {
+        return timestep > m_timestep;
+        }
+
+    /// Get the point where triggering will stop
+    uint64_t getTimestep() {return m_timestep;}
+
+    /// Set the point where triggering will stop
+    void setTimestep(uint64_t timestep) {m_timestep = timestep;}
+
+    protected:
+        /// Always trigger timestep this timestep
+        uint64_t m_timestep;
+    };
+
+/** Not trigger
+ *
+ *  Negates any given trigger.
+*/
+class PYBIND11_EXPORT NotTrigger : public Trigger
+    {
+    public:
+        NotTrigger(std::shared_ptr<Trigger> trigger) :
+            Trigger(), m_trigger(trigger) {}
+
+        bool compute(uint64_t timestep)
+            {
+            return !(m_trigger->operator()(timestep));
```

Comment:
If you think it is cleaner, you can also use:
```
(*m_trigger)(timestep)
```
I don't have a preference.

Reply:
I don't either. Your suggestion is a bit shorter, but I think either is clear. I would vote to keep as is unless you or @bdice feel strongly about one of the styles.